internal/provisioner/plain/plain.go

```diff
@@ -50,6 +62,15 @@ func ValidateBundle(fsys fs.FS) error {
 	return nil
 }
 
+func isRegistryV1Bundle(bd *rukpakv1alpha2.BundleDeployment) (bool, error) {
+	// Should not happen as it would have been as a nil pointer error earlier in the
+	// reconciler.
+	if bd == nil {
+		return false, fmt.Errorf("unable to find bundle format through provisioner")
+	}
+	return bd.Spec.ProvisionerClassName == registryv1.ProvisionerID, nil
```

Comment:
It's a bit confusing, IMO, to have a "plain" provisioner that handles multiple provisioner IDs and switches its logic based on that.

I think we should do one of the following:
1. Have two separate handler functions: one for `plain` and one for `registry` (and they can share any shared logic via a library if necessary)
2. Have a single handler with a single provisioner ID that introspects the bundle contents and if it looks like a registry+v1 bundle, just treat it that way. Otherwise, assume `plain`.

Reply:
Using option (1) for now, to make less user-facing changes.